### PR TITLE
Add support for Postgres citext

### LINKED ISSUE #1847: Postgres citext not supported
[Postgres citext string type](https://www.postgresql.org/docs/current/static/citext.html) has been a standard extension since [8.4](https://www.postgresql.org/docs/8.4/static/citext.html). This is similar to the support `text` type, except that comparisons are case insensitive.

As it stands now, model builder will default to String type, but fail to recognize `citext` as a varying type, so it leaves `O.Length(Integer.MAX_VALUE)` in the length calculation:

```scala
override def length: Option[Int] = {
  val l = super.length
  if(tpe == "String" && varying && l == Some(2147483647)) None
  else l
}
```

Note that `text` is explicitly supported by the postgres JDBC driver, which will output sql type `VARCHAR` and length `2147483647`. `citext` gets sql type `OTHER`.

### PR DESCRIPTION
**Pull request by stewSquared**
Resolves #1847 
Note that this PR doesn't handle handling of default values. See #1842

### CODE REVIEW

File: slick/src/main/scala/slick/jdbc/JdbcModelBuilder.scala [hunk 1]
```diff
@@ -190,15 +190,17 @@ class JdbcModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(imp
...
     /** Column length of string types */
     def length: Option[Int] = if(tpe == "String") meta.size else None // Only valid for strings!
     /** Indicates whether this should be a varchar in case of a string column.
-      * Currently defaults to true. Should be based on the value of dbType in the future. */
-    def varying: Boolean =
-      Seq(java.sql.Types.NVARCHAR, java.sql.Types.VARCHAR, java.sql.Types.LONGVARCHAR, java.sql.Types.LONGNVARCHAR) contains meta.sqlType
+      * Should be based on the value of dbType in the future. */
+    def varying: Boolean = {
+      import java.sql.Types._
+      Seq(NVARCHAR, VARCHAR, LONGVARCHAR, LONGNVARCHAR) contains meta.sqlType
```
Comment (stewSquared) [inline slick/src/main/scala/slick/jdbc/JdbcModelBuilder.scala]: Note that postgres jdbc driver maps `text` to `VARCHAR`, but `citext` to `OTHER`, which is why `text` type doesn't need special treatment in `def varying` below.

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version=0.13.13
```
Comment (stewSquared) [inline project/build.properties]: I needed to upgrade sbt version for this fix: https://github.com/sbt/sbt/pull/3265

There is a platform-specific issue that comes up for some people with newer versions of ncurses: https://github.com/sbt/sbt/issues/3240

Let me know if this should be part of a separate PR.

**Comment by hvesalai** [comment]
@stewSquared could you rebase this to current master.

**Comment by stewSquared** [comment]
@hvesalai Sorry for the late reply.
I'll spruce up my pulls and issues early next week.
